Documentation/Signoffs/OEM Component- Auto_Light-signoff.md

```diff
@@ -0,0 +1,99 @@
+
+### Specification & constraints
+   - Must detect high beam,low beam, left indicator, and right indicator
+   - Must use OEM componets as the inputs
+   - The signal input must not exceed 5 V 
+   - Implement a way to protect the signal from the input voltage varying from 10 V to 15 V
+   - Current to the signal must be less than 1 percent of the current to the light
+   - Prevent interference among signals through isolation methods
+   
+# Auto Light Wiring schematic
+![opel2437](https://user-images.githubusercontent.com/101354063/205468261-7fca3396-009b-4eb8-9960-962065d22ac8.png)
+### Figure 1: Headlamp(High and low Relay)
+<br/>
+
+## Design: A Voltage divider from Auto Light to the Microcontroller
+<br/>
+
+![image](https://user-images.githubusercontent.com/101354063/206637205-1bb47b94-0cef-4220-b707-90ca1036a34e.png)
+
+### Figure 2: Voltage Divder with Opto-isolation
+
+Description
+-------
+### Auto Light Wiring Diagram
+The auto light wiring diagram shows the whole wiring for each component in the headlight of the Nissan vehicle. However, the main focus are the headlamp relays shown in Figure 1 and they link to the headlamps. This displays how the low beam and high beam functions operate and make the headlight indicate. The wiring diagram reveals that the battery is attached to the CPU through the relay showing that the relay receives power from the battery, but the current is operated by the CPU. The input from the battery is also connected to one side of the relay switch. This means that the input to the lamp and headlight should be 12 V. Also, Figure 1 shows that their fuse connects to the output of the relay, and the fuse can take up to 10 amps. Overall, the key point is to see the headlamp relay, what is connected, and how it works. 
+
+
+### Design Voltage 
+From the information in Auto Light Wiring diagram, a method to detect the output of the relay can be created. The design shown in Figure 2 is to splice and add a voltage divider to connect to the microcontroller. This process uses the 12 V potential that is sent to the lamp to illuminate the light to trigger a reaction on the microcontroller. 
+
+The design behind the voltage divider is to reduce the auto light voltage to no more than 5 V and have a current of less than 1% of the current to the light. The current through the light is 4.79 A which makes the limit to the voltage divider 47.9 mA. 40 mA is decided to be a limit for the safety of the devices added and not exceed the constraint. Through Ohm's law, the total resistance is found to be approximately 375 Ohms assuming that the input voltage could vary from 10 to 15 V. This achieves a maximum of 5 V from a 15 V input between the two series resistors of 250 Ohms and 125 Ohms respectively. The average active voltage applied should be approximately 12 V. Thus, the limit should not be exceeded. A 40 mA fuse will be installed to protect against any extreme voltage spikes outside of the 10 to 15 V range.
```

Comment:
I agree with the use of current limitation, but specifically why is it being included?